### ISSUE TITLE
Python wheel does not work with Pip 20.x

### ISSUE THREAD

**Issue by itzikjan** [dmlc/xgboost#5221]
We are installing xboost with python 2 and python 3.6  (on EMR)
Since today morning the installation step failed in all our EMRs (trying using python 2 and python 3).
1.	sudo /usr/local/bin/pip install xgboost==0.90
ERROR: Could not find a version that satisfies the requirement xgboost==0.90 (from versions: 0.4a12, 0.4a13, 0.4a14, 0.4a15, 0.4a18, 0.4a19, 0.4a20, 0.4a21, 0.4a22, 0.4a23, 0.4a24, 0.4a25, 0.4a26, 0.4a27, 0.4a28, 0.4a29, 0.4a30, 0.6a1, 0.6a2, 0.7.post3, 0.7.post4, 0.71, 0.72.1, 0.80, 0.81, 0.82)
ERROR: No matching distribution found for xgboost==0.90

2.	sudo /usr/local/bin/pip install xgboost
ollecting xgboost
  Using cached xgboost-0.82.tar.gz (665 kB)
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/mnt/tmp/pip-install-eyulfa/xgboost/setup.py'"'"'; __file__='"'"'/mnt/tmp/pip-install-eyulfa/xgboost/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /mnt/tmp/pip-install-eyulfa/xgboost/pip-egg-info
         cwd: /mnt/tmp/pip-install-eyulfa/xgboost/

**Comment by trivialfis** [comment]
@hcho3 Could you please help taking a look?

**Comment by StrikerRUS** [comment]
@itzikjan Please take a look at https://github.com/microsoft/LightGBM/issues/2702#issuecomment-576961045. I guess it is the same problem.

**Comment by itzikjan** [comment]
On 21/01/2020 pip was updated from 19.3.1 to 20.X.
The installation of xgboost fails with pip version 20.X.
Using pip 19.3.1 instead of 20.X, no problem installing xgboost

**Comment by hcho3** [comment]
I wonder why pip 20.X is causing a problem. We should investigate.

**Comment by hcho3** [comment]
@dmlc/xgboost-committer We may want to release 1.0 soon, as pip 20.x is breaking many existing Python wheels.

Examples:
* LightGBM: https://github.com/microsoft/LightGBM/issues/2702#issuecomment-576961045
* MXNet: https://lists.apache.org/thread.html/r5e82ed14ee44bb959df47af77d32b66da632716109657801c20d7ded%40%3Cdev.mxnet.apache.org%3E

**Comment by hcho3** [comment]
Related:
* https://github.com/microsoft/LightGBM/issues/2707
* pypa/pip#7636
* pypa/pip#7626

**Comment by trivialfis** [comment]
Yup.  There are 2 blocking PRs, first is the scikit learn wrapper waiting for review another is io for metainfo.  I'm not sure how is rabit doing.

**Comment by StrikerRUS** [comment]
Inspiring a hope post from pip maintainer:
> pradyunsg commented 3 hours ago
Expect a bugfix release for this tomorrow, assuming I recover from my current headache by then. :)
https://github.com/pypa/pip/issues/7626#issuecomment-577860648

**Comment by trivialfis** [comment]
That's great news, thanks!

**Comment by hcho3** [comment]
The issue is now fixed in latest pip (20.0.2).

Example:
```
(pip20) ubuntu@ip-172-31-1-97:~$ pip install --force-reinstall pip==20.0.2
Collecting pip==20.0.2
  Downloading pip-20.0.2-py2.py3-none-any.whl (1.4 MB)
     |████████████████████████████████| 1.4 MB 4.3 MB/s
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.0.1
    Uninstalling pip-20.0.1:
      Successfully uninstalled pip-20.0.1
Successfully installed pip-20.0.2
(pip20) ubuntu@ip-172-31-1-97:~$ python -m pip install xgboost
Collecting xgboost
  Using cached xgboost-0.90-py2.py3-none-manylinux1_x86_64.whl (142.8 MB)
Collecting scipy
  Using cached scipy-1.4.1-cp37-cp37m-manylinux1_x86_64.whl (26.1 MB)
Collecting numpy
  Using cached numpy-1.18.1-cp37-cp37m-manylinux1_x86_64.whl (20.1 MB)
Installing collected packages: numpy, scipy, xgboost
Successfully installed numpy-1.18.1 scipy-1.4.1 xgboost-0.90
```
